mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
Is it really worth to have this caching causing more gen1 pressure and causing hard to debug issues when user code still wants to use original reference